disk-utils/mkswap.c

```diff
@@ -469,7 +474,9 @@ static void write_header_to_device(struct mkswap_control *ctl)
 	assert(ctl->fd > -1);
 	assert(ctl->signature_page);
 
-	if (lseek(ctl->fd, SIGNATURE_OFFSET, SEEK_SET) != SIGNATURE_OFFSET)
+	off_t offset = SIGNATURE_OFFSET + ctl->offset;
```

Comment:
Please, please, define the variable at the beginning of the function ;-) Thanks.